src/metadata_validator.py

```diff
@@ -159,25 +159,28 @@ def validate_nodes(self, data_records):
         validated_count = 0
         try:
             for record in data_records:
-                qc_result = get_qc_result(record, self.submission, VALIDATION_TYPE_METADATA, self.mongo_dao)
+                qc_result = None
+                if record.get(QC_RESULT_ID):
+                    qc_result = self.mongo_dao.get_qcRecord(record[QC_RESULT_ID])
                 status, errors, warnings = self.validate_node(record)
+                if errors and warnings and not qc_result:
```

Comment:
Looks like only if there are both errors and warnings a new qcResult will be created? Will it create one when only errors or only warnings exist?

Reply:
Updated